src/java/boa/evaluator/BoaEvaluator.java

```diff
@@ -16,26 +16,159 @@
  */
 package boa.evaluator;
 
+import java.io.File;
 import java.io.IOException;
-import org.apache.commons.cli.CommandLine;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+
 import org.apache.commons.cli.HelpFormatter;
 import org.apache.commons.cli.Options;
-import org.apache.commons.cli.PosixParser;
+import org.apache.commons.io.FileUtils;
+
+import boa.compiler.BoaCompiler;
+import boa.datagen.DefaultProperties;
+import boa.datagen.util.FileIO;
 
 /**
  * The main entry point for Boa REPL.
  *
  * @author hridesh
- * 
+ *
  */
-public class BoaEvaluator {	
-	public static void main(final String[] args) throws IOException {
-	    System.out.println("TODO: integrate existing evaluator code here.");
-	}	
-	private static final void printHelp (Options options, String message) {
-	    	String header = "The most commonly used Boa options are:";
-	    	String footer = "\nPlease report issues at http://www.github.com/boalang/";
-	    	System.err.println(message);
+public class BoaEvaluator {
+
+	private final String PROG_PATH;
+	private final String DATA_PATH;
+	private final String COMPILATION_DIR;
+	private  String OUTPUT_DIR;
+
+	public BoaEvaluator(String prog, String data) {
+		this.PROG_PATH = prog;
+		this.DATA_PATH = data;
+		this.COMPILATION_DIR = "./compile";
+		this.OUTPUT_DIR =  System.getProperty("java.io.tmpdir");
+		setup();
+	}
+
+	public BoaEvaluator(String prog, String data, String outDir){
+		this(prog, data);
+		this.OUTPUT_DIR = outDir;
+		setup();
+	}
+
+	private static final void printHelp(Options options, String message) {
+		String header = "The most commonly used Boa options are:";
+		String footer = "\nPlease report issues at http://www.github.com/boalang/";
+		System.err.println(message);
 		new HelpFormatter().printHelp("Boa", header, options, footer);
 	}
-}
+
+	public void evaluate() {
+		final String[] actualArgs = createHadoopProgramArguments();
+		File srcDir = new File(this.COMPILATION_DIR);
+		try {
+			URL srcDirUrl = srcDir.toURI().toURL();
+
+			ClassLoader cl = new URLClassLoader(new URL[] { srcDirUrl }, ClassLoader.getSystemClassLoader());
+			Class<?> cls = cl.loadClass(getClassNameForGeneratedJavaProg());
+			final Method method = cls.getMethod("main", String[].class);
+			method.invoke(null, (Object) actualArgs);
+
+		} catch (Throwable exc) {
+			exc.printStackTrace();
+		}
+	}
+
+	public static void main(final String[] args) {
+
+		if (args.length < 2 || args.length > 3) {
```

Comment:
The command line options ought to be processed in a uniform way. First general options to select Boa components, then component-specific options. See the structure of the main function in other components such as the data generation component.